sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -198,9 +198,15 @@ public void SetProperty(string name, object value)
         /// </summary>
         public void Dispose()
         {
-            Request?.Dispose();
-            _response?.Dispose();
+            Request.Dispose();
```

Comment:
In the prior version, we checked this for null.  Could Request ever be null here, would it make sense to keep that check?

Reply:
No, we don't expect it to be null. Probably, we should add an explicit check in constructor